juspay/hyperswitch
# fix: [mollie] locale validation irrespective of auth type

## Type of Change



## Description
Irrespective of 3ds or non 3ds locale should be set from browser_info language
Add Hyperswitch reference in the payment and refunds request metadata


### Additional Changes




## How did you test it?
Manual testing:
1. Create payment with authentication_type as "no_three_ds" with "browser_info.language" in the payment request.Payment should succeed 
`{
    "amount": 100000,
    "currency": "EUR",
    "capture_method": "automatic",
    "amount_to_capture": 1000,
    "customer_id": "StripeCustomer",
    "email": "arjun.karthik@gmail.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://google.com",
    "confirm": true,
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "378282246310005",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "1234"
        }
    },
    "browser_info": {
        "language": "en_US"
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "zip": "94122",
            "country": "US"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "zip": "94122",
            "country": "US"
        }
    }
}`
2. Create payment with authentication_type as "three_ds" with "browser_info.language" in the payment request.Payment should succeed
`{
    "amount": 100000,
    "currency": "EUR",
    "capture_method": "automatic",
    "amount_to_capture": 1000,
    "customer_id": "StripeCustomer",
    "email": "arjun.karthik@gmail.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    "authentication_type": "three_ds",
    "return_url": "https://google.com",
    "confirm": true,
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "378282246310005",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "1234"
        }
    },
    "browser_info": {
        "language": "en_US"
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "zip": "94122",
            "country": "US"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "zip": "94122",
            "country": "US"
        }
    }
}`
4. Create payment with authentication_type as "no_three_ds" with out "browser_info" in the payment request. Payment will fail with 400
`{
    "amount": 100000,
    "currency": "EUR",
    "capture_method": "automatic",
    "amount_to_capture": 1000,
    "customer_id": "StripeCustomer",
    "email": "arjun.karthik@gmail.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://google.com",
    "confirm": true,
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "378282246310005",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "1234"
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "zip": "94122",
            "country": "US"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "zip": "94122",
            "country": "US"
        }
    }
}`
6. Create payment with authentication_type as "three_ds" with out "browser_info" in the payment request. Payment will fail with 400
`{
    "amount": 100000,
    "currency": "EUR",
    "capture_method": "automatic",
    "amount_to_capture": 1000,
    "customer_id": "StripeCustomer",
    "email": "arjun.karthik@gmail.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    "authentication_type": "three_ds",
    "return_url": "https://google.com",
    "confirm": true,
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "378282246310005",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "1234"
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "zip": "94122",
            "country": "US"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "zip": "94122",
            "country": "US"
        }
    }
}`


## Checklist
